File: week-06/restaurant-bot/activate-n8n.js
const sqlite3 = require('sqlite3').verbose();
const db = new sqlite3.Database('/home/node/.n8n/database.sqlite');

// Find workflow tables
db.all("SELECT name FROM sqlite_master WHERE type='table'", (err, tables) => {
  if (err) { console.log('Error:', err.message); db.close(); return; }
  const wfTables = tables.filter(t => t.name && t.name.toLowerCase().includes('workflow'));
  console.log('Workflow tables:', wfTables.map(t => t.name));
  
  // Check which table has our workflow
  for (const tableName of wfTables) {
    db.all(`SELECT * FROM "${tableName}" WHERE name LIKE '%SpiceRoute%'`, (err, rows) => {
      if (err) { console.log('Query error:', err.message); return; }
      if (rows.length > 0) {
        console.log('Found workflow:', rows[0].name);
        console.log('Current active:', rows[0].active);
        // Activate it
        db.run(`UPDATE "${tableName}" SET active = 1 WHERE name LIKE '%SpiceRoute%'`);
        console.log('Activated!');
        db.close();
      }
    });
  }
});

File: week-06/restaurant-bot/activate-n8n.py
import sqlite3

db = sqlite3.connect('/home/huzaifa/.local/share/containers/storage/volumes/n8n-restaurant-data/_data/database.sqlite')
cursor = db.cursor()

cursor.execute("SELECT name FROM sqlite_master WHERE type='table'")
tables = [t[0] for t in cursor.fetchall()]
print('All tables:', tables)

wf_tables = [t for t in tables if 'workflow' in t.lower() or 'entity' in t.lower()]
print('Workflow tables:', wf_tables)

for t in wf_tables:
    cursor.execute(f'SELECT id, name, active FROM "{t}" WHERE name LIKE "%SpiceRoute%"')
    rows = cursor.fetchall()
    if rows:
        print(f'Found in {t}: id={rows[0][0]}, name={rows[0][1]}, active={rows[0][2]}')
        cursor.execute(f'UPDATE "{t}" SET active = 1 WHERE name LIKE "%SpiceRoute%"')
        db.commit()
        print('Activated!')
        cursor.execute(f'SELECT id, name, active FROM "{t}" WHERE name LIKE "%SpiceRoute%"')
        print('After:', cursor.fetchall())
        break

db.close()
